test/puppetlabs/puppetdb/cheshire_test.clj

```diff
@@ -35,6 +36,14 @@
       (is (= (.toString sw)
              "{\n  \"a\" : 1,\n  \"b\" : 2\n}")))))
 
+(deftest test-date-generation
+  (let [t (java.util.Date. 5928174905781)]
```

Comment:
This is just checks that our generate-string generates the same (expected) output for both a java.util.Date and a joda DateTime.   Before, Date (specifically), didn't work.  And now that you mention it, perhaps we should check Timestamp here too, since that's one of our important types.  I'll add it in a minute.  You can see the failure by changing generate-string to not include default-opts in the core/generate-string calls.